src/ApplicationInsights.ServiceFabric.Native/Net45/ServiceRemotingMessageHeadersExtensions.cs

```diff
@@ -1,13 +1,63 @@
 ﻿namespace Microsoft.ApplicationInsights.ServiceFabric.Remoting.Activities
 {
+    using Microsoft.ServiceFabric.Actors.Remoting.Builder;
     using Microsoft.ServiceFabric.Services.Remoting;
+    using System;
+    using System.Reflection;
     using System.Text;
 
     /// <summary>
     /// Class with extension helper methods for working with headers in a service remoting message
     /// </summary>
     internal static class ServiceRemotingMessageHeadersExtensions
     {
+        private static class ActorMessageHeadersHelper
+        {
+            private static Func<byte[], object> deserializeMethodDelegate;
+            private static FieldInfo interfaceIdField;
+            private static FieldInfo methodIdField;
+
+            static ActorMessageHeadersHelper()
+            {
+                try
+                {
+                    Type actorMessageHeadersType = typeof(ActorMethodDispatcherBase).Assembly.GetType("Microsoft.ServiceFabric.Actors.Remoting.ActorMessageHeaders");
+                    MethodInfo method = actorMessageHeadersType?.GetMethod("Deserialize", BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic);
+
+                    if (method != null)
+                    {
+                        deserializeMethodDelegate = (Func<byte[], object>)Delegate.CreateDelegate(typeof(Func<byte[], object>), method);
+                    }
+
+                    interfaceIdField = actorMessageHeadersType?.GetField("InterfaceId", BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic);
+                    methodIdField = actorMessageHeadersType?.GetField("MethodId", BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic);
+                }
+                catch (Exception)
+                {
+                    // Can't let the static constructor brings down the process if anything happens.
+                    // At worst, we just don't get any method namese.
+                }
+            }
+            
+            public static bool GetIds(byte[] headerBytes, out int methodId, out int interfaceId)
```

Comment:
c# pattern for naming such methods is TryGetIds

Reply:
Agree